src/order_executer/service/data_service.py

```diff
@@ -34,28 +34,40 @@ def get_channel_data(self, start, end):
 
 
 class TrainingDataService:
-    def __init__(self, interval_period_in_minutes, channels, start_time, end_time):
+    def __init__(self, interval_period_in_minutes, channels, start_date, end_date):
+        """
+        Tries loading saved data from disk if possible. Otherwise, scrapes it first.
+        The starting exact time will be 00:00 of the requested start date and 23:59 of the end date.
+        :param interval_period_in_minutes: Time interval between data samples
+        :param channels: Data scrapers of choice
+        :param start_date: YYYY-MM-DD date written as string
+        :param end_date: YYYY-MM-DD date written as string
+        """
         self.interval_period_in_minutes = interval_period_in_minutes
         self.channels = channels
-        self.start_time = time_helpers.str_to_timestamp(start_time, format="date")
-        self.end_time = time_helpers.str_to_timestamp(end_time, format="date")
+        self.start_timestamp = time_helpers.str_to_timestamp(start_date, format='date')
+        self.start_timestamp = time_helpers.get_start_of_the_day(self.start_timestamp)  # Start will be at 00:00
+
+        self.end_timestamp = time_helpers.str_to_timestamp(end_date, format='date')
+        self.end_timestamp = time_helpers.get_end_of_the_day(self.end_timestamp)
+        self.end_timestamp = time_helpers.adjust_last_possible_timestamp(self.end_timestamp)  # End will be at 23:59
 
         # First timestamp
-        self.last_end_time = start_time
+        self.last_end_timestamp = self.start_timestamp
 
     def initialize(self):
         """
         Gets data between the start timestamp and the lookback configured.
         :return: Dictionary with data for each channel
         """
         print("Initializing Training Data Service")
-        window_before_start_time = time_helpers.get_production_start_timestamp(self.start_time)
+        window_after_start_time = self.start_timestamp + (1000 * 60 * config.LATEST_DATA_LOOKBACK_MIN)
```

Comment:
Now I add the lookback to the start time instead of subtracting it to make the training data start from 00:00 and avoid incomplete days of data